src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -227,6 +234,17 @@ public void delete(final String eventTypeName, final String timelineId, final Cl
         }
     }
 
+    public void deleteAllTimelinesForEventType(final String eventTypeName)
+            throws TopicDeletionException, TimelineException, NotFoundException {
+        LOG.info("Deleting all timelines for event type {}", eventTypeName);
+        final List<Timeline> timelines = getTimelines(eventTypeName);
+        for (final Timeline timeline:timelines) {
+            final TopicRepository topicRepository = getTopicRepositoryForTimeline(timeline);
+            topicRepository.deleteTopic(timeline.getTopic());
```

Comment:
As you saw from my PR - topic can already be removed for old timeline. I think you should handle here the case that topic doesn't exist and continue with removal of next timeline (but not interrupt the whole process).